README.md

```diff
@@ -1,2 +1,437 @@
+* [Common packages](#common-packages)
+* [Directories and files](#directories-and-files)
+* [Imports](#imports)
+* [JSX](#jsx)
+* [Whitespace](#whitespace)
+* [Methods](#methods)
+* [Common Patterns](#common-patterns)
+
+## Javascript
+
+* For general Javascript syntax see [this styleguide](https://github.com/airbnb/javascript) except for the following cases:
+
+* Always use parenthesis around single parameter arrow functions.
+  ```js
+  // good
+  cars.filter((car) => car.color == 'red');
+
+  // bad
+  cars.filter(car => car.color == 'red');
+  ```
+
+* Object whitespace
+  ```js
+  // good
+  const car = {color: 'red', model: 'Saturn'};
+
+  // bad
+  const car = { color: 'red', model: 'Saturn' };
+  // bad
+  const car = {color:'red',model:'Saturn'};
+  ```
+
+
+## Common packages
+
+## Directories and files
+
+### Directory structure
+
+### Filenames
+
+* Use camelCase that matches the default export (this makes it easier to edit in a text editor)
+
+## Imports
+
+* Import modules in order of more general to more specific:
+  - Utility libraries (e.g. lodash)
+  - Frameworks (e.g. Redux)
+  - Utility functions/constants
+  - Actions
+  - Components
+
+* When there are more than 5 import statements, add blank lines between groups.
+
+  ```js
+  import { connect } from 'react-redux';
+  import { Link } from 'react-router-dom';
+  import { Card, Button } from 'semantic-ui-react';
+  import { fetchUsers } from 'actions/users';
+  import searchBy from 'utils/search';
+
+  import Search from 'components/helpers/Search';
+  import Page from 'components/layouts/Page';
+  ```
+
+## JSX
+
+JSX leverages syntactic similarity with HTML so similar conventions should be observed.
+
+* rendering
+  ```js
+  // good
+  render() {
+    return <Button text="Save" onChange={this.save}/>;
+  }
+  // good
+  render() {
+    return (
+      <div className="action-bar">
+        <Button text="Save" onChange={this.save}/>
+        <Button text="Cancel" onChange={this.goBack}/>
+      </div>
+    );
+  }
+
+  // bad
+  render() {
+    return <div className="action-bar">
+      <Button text="Save" onChange={this.save}/>
+      <Button text="Cancel" onChange={this.goBack}/>
+    </div>;
+  }
+  ```
+
+* use double quotes for attributes
+  ```js
+  // good
+  <Button text="Save" onChange={this.save}/>
+
+  // bad
+  <Button text='Save' onChange={this.save}/>
+  ```
+
+* Do not use braces around double quotes
+  ```js
+  // good
+  <Button text="Save" onChange={this.save}/>
+
+  // bad
+  <Button text={"Save"} onChange={this.save}/>
+  ```
+
+* Do not use spaces around =
+  ```js
+  // good
+  <Button text="Save" onChange={this.save}/>
+
+  // bad
+  <Button text = "Save" onChange = {this.save}/>
+  ```
+
+* Inline short components
+  ```js
+  // good
+  <Button text="Save" onChange={this.save}/>
+
+  // bad
+  <Button
+    text="Save"
+    onChange={this.save}/>
+  ```
+
+* Multiline each prop on it's own line
+  ```js
+  // good
+  <Input
+    type="email"
```

Comment:
I have one really big reason why I hate this: it's a pain to deal with in an editor.

Related: several of my suggestions in this guide are because they make it easier to use sublime/atom